src/Elliptic/Systems/Poisson/FirstOrderSystem.hpp

```diff
@@ -84,18 +87,18 @@ struct FirstOrderSystem {
   using fields_tag =
       ::Tags::Variables<tmpl::append<primal_fields, auxiliary_fields>>;
 
-  using fluxes =
-      tmpl::conditional_t<BackgroundGeometry == Geometry::Euclidean,
-                          EuclideanFluxes<Dim>, NonEuclideanFluxes<Dim>>;
-  using sources = Sources;
+  using fluxes = Fluxes<Dim, BackgroundGeometry>;
+  using sources = Sources<Dim, BackgroundGeometry>;
 
   // The tag of the operator to compute magnitudes on the manifold, e.g. to
   // normalize vectors on the faces of an element
+  using inv_metric_tag = tmpl::conditional_t<
+      BackgroundGeometry == Geometry::Euclidean, void,
+      gr::Tags::InverseSpatialMetric<Dim, Frame::Inertial, DataVector>>;
```

Comment:
Should there be a sanity check to make sure that Geometry has precisely two choices?

Reply:
I think that would be necessary if the "else"-clause implied a particular enum case, but here the "else" just means: For anything that's not `FlatCartesian`, use the spatial metric tag. It would keep working even if extra enum cases were added.